rel/env.sh.eex

```diff
@@ -1,7 +1,37 @@
-# release always starts in a distributed mode
-# but to provide unified way of configuring Jellyfish
-# distribution both in the local and production environment
-# we introduce our own env vars and use them to
-# set RELASE_NODE and RELASE_COOKIE
-export RELEASE_NODE=${JF_NODE_NAME:-${RELEASE_NODE:-jellyfish}}
-export RELEASE_COOKIE=${JF_COOKIE:-${RELEASE_COOKIE:-"$(cat "$RELEASE_ROOT/releases/COOKIE")"}}
\ No newline at end of file
+# We introduce our own env vars and use them to set
+# RELEASE_NODE and RELEASE_COOKIE.
+# This is to provide a unified way of configuring Jellyfish distribution
+# in both development and production environments
+export RELEASE_NODE=${JF_DIST_NODE_NAME:-${RELEASE_NODE:-jellyfish}}
+export RELEASE_COOKIE=${JF_DIST_COOKIE:-${RELEASE_COOKIE:-jellyfish_cookie}}
+
+# Use longnames by default.
+# They are required to form a cluster across diffrent machines.
+# In development, we also use longnames (when starting a new node).
+export RELEASE_DISTRIBUTION=${JF_DIST_MODE:-${RELEASE_DISTRIBUTION:-name}}
+
+# We only set min and max ports for start and daemon commands.
+# In other case, when someone wants to expose only one port 
+# (JF_DIST_MIN_PORT==JF_DIST_MAX_PORT), we won't be able to 
+# connect to already running node with the `remote` command.
+if [ "$JF_DIST_MIN_PORT" != "" ] ; then
+    case $RELEASE_COMMAND in
+    start*|daemon*)
+        ELIXIR_ERL_OPTIONS="$ELIXIR_ERL_OPTIONS -kernel inet_dist_listen_min $JF_DIST_MIN_PORT"
+        export ELIXIR_ERL_OPTIONS
+        ;;
+    *)
+        ;;
+    esac
+fi
+
+if [ "$JF_DIST_MAX_PORT" != "" ] ; then
+    case $RELEASE_COMMAND in
+    start*|daemon*)
+        ELIXIR_ERL_OPTIONS="$ELIXIR_ERL_OPTIONS -kernel inet_dist_listen_max $JF_DIST_MAX_PORT"
+        export ELIXIR_ERL_OPTIONS
+        ;;
+    *)
+        ;;
+    esac
+fi
```

Comment:
I am unsure about `eex` syntax, but don't you set `MIN_PORT` or `MAX_PORT`? Because in the second, if you override the value from the first if?

Reply:
Nope, this should append to the already existing ELIXIR_ERL_OPTIONS variable

In case of setting both MIN and MAX port we will have `-kernel inet_dist_listen_min 2222 -kernel inet_dist_listen_max 3333`